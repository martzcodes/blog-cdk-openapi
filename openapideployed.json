{
  "openapi" : "3.0.1",
  "info" : {
    "title" : "OpenApi",
    "version" : "2020-11-03T18:05:43Z"
  },
  "servers" : [ {
    "url" : "https://c638jrlvt0.execute-api.us-east-1.amazonaws.com/{basePath}",
    "variables" : {
      "basePath" : {
        "default" : "/prod"
      }
    }
  } ],
  "paths" : {
    "/{hello}/advanced" : {
      "post" : {
        "parameters" : [ {
          "name" : "hello",
          "in" : "path",
          "required" : true,
          "schema" : {
            "type" : "string"
          }
        } ],
        "requestBody" : {
          "content" : {
            "application/json" : {
              "schema" : {
                "$ref" : "#/components/schemas/AdvancedModel"
              }
            }
          },
          "required" : true
        },
        "responses" : {
          "200" : {
            "description" : "200 response",
            "headers" : {
              "Access-Control-Allow-Origin" : {
                "schema" : {
                  "type" : "string"
                }
              },
              "Access-Control-Allow-Credentials" : {
                "schema" : {
                  "type" : "string"
                }
              },
              "Content-Type" : {
                "schema" : {
                  "type" : "string"
                }
              }
            },
            "content" : {
              "application/json" : {
                "schema" : {
                  "$ref" : "#/components/schemas/ResponseModel"
                }
              }
            }
          }
        },
        "security" : [ {
          "blogcdkopenapiblogAuthorizer0C135D8A" : [ ]
        } ]
      }
    },
    "/{hello}/basic" : {
      "post" : {
        "parameters" : [ {
          "name" : "hello",
          "in" : "path",
          "required" : true,
          "schema" : {
            "type" : "string"
          }
        } ],
        "requestBody" : {
          "content" : {
            "application/json" : {
              "schema" : {
                "$ref" : "#/components/schemas/BasicModel"
              }
            }
          },
          "required" : true
        },
        "responses" : {
          "200" : {
            "description" : "200 response",
            "headers" : {
              "Access-Control-Allow-Origin" : {
                "schema" : {
                  "type" : "string"
                }
              },
              "Access-Control-Allow-Credentials" : {
                "schema" : {
                  "type" : "string"
                }
              },
              "Content-Type" : {
                "schema" : {
                  "type" : "string"
                }
              }
            },
            "content" : {
              "application/json" : {
                "schema" : {
                  "$ref" : "#/components/schemas/ResponseModel"
                }
              }
            }
          }
        },
        "security" : [ {
          "blogcdkopenapiblogAuthorizer0C135D8A" : [ ]
        } ]
      }
    }
  },
  "components" : {
    "schemas" : {
      "BasicModel" : {
        "title" : "BasicModel",
        "required" : [ "someNumber", "someString" ],
        "type" : "object",
        "properties" : {
          "someNumber" : {
            "type" : "number"
          },
          "someString" : {
            "type" : "string"
          }
        },
        "additionalProperties" : false
      },
      "ResponseModel" : {
        "title" : "ResponseModel",
        "required" : [ "message" ],
        "type" : "object",
        "properties" : {
          "message" : {
            "type" : "string"
          }
        },
        "additionalProperties" : false
      },
      "AdvancedModel" : {
        "title" : "AdvancedModel",
        "required" : [ "basic", "greeting" ],
        "type" : "object",
        "properties" : {
          "greeting" : {
            "type" : "string"
          },
          "postfix" : {
            "type" : "string"
          },
          "basic" : {
            "$ref" : "#/components/schemas/BasicModel"
          }
        },
        "additionalProperties" : false
      }
    },
    "securitySchemes" : {
      "blogcdkopenapiblogAuthorizer0C135D8A" : {
        "type" : "apiKey",
        "name" : "Authorization",
        "in" : "header",
        "x-amazon-apigateway-authtype" : "custom"
      }
    }
  }
}