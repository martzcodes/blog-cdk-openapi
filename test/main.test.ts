import '@aws-cdk/assert/jest';
import { App } from '@aws-cdk/core';
import * as deployedApiSpec from '../openapideployed.json';
import { MyStack } from '../src/main';

test('Snapshot', () => {
  const app = new App();
  const stack = new MyStack(app, 'test');

  expect(stack).not.toHaveResource('AWS::S3::Bucket');
  expect(app.synth().getStackArtifact(stack.artifactId).template).toMatchSnapshot();
});

test('Api Spec', () => {
  const app = new App();
  const stack = new MyStack(app, 'test', { generateApiSpec: true });
  console.log(`Stack: ${JSON.stringify(stack.apiSpec)}`);
  console.log(`deployed: ${JSON.stringify(deployedApiSpec)}`);
  expect(stack.apiSpec).toMatchSnapshot({
    info: {
      version: expect.any(String),
    },
  });
});
