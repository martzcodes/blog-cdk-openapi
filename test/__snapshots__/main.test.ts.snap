// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api Spec 1`] = `
Object {
  "components": Object {
    "schemas": Object {
      "AdvancedModel": Object {
        "properties": Object {
          "basic": Object {
            "$ref": "#/components/schemas/BasicModel",
          },
          "greeting": Object {
            "type": "string",
          },
          "postfix": Object {
            "type": "string",
          },
        },
        "required": Array [
          "greeting",
          "basic",
        ],
        "title": "AdvancedModel",
        "type": "object",
      },
      "BasicModel": Object {
        "properties": Object {
          "someNumber": Object {
            "type": "number",
          },
          "someString": Object {
            "type": "string",
          },
        },
        "required": Array [
          "someString",
          "someNumber",
        ],
        "title": "BasicModel",
        "type": "object",
      },
      "ResponseModel": Object {
        "properties": Object {
          "message": Object {
            "type": "string",
          },
        },
        "required": Array [
          "message",
        ],
        "title": "ResponseModel",
        "type": "object",
      },
    },
    "securitySchemes": Object {
      "authorizer": Object {
        "in": "header",
        "name": "Authorization",
        "type": "apiKey",
        "x-amazon-apigateway-authtype": "custom",
      },
    },
  },
  "info": Object {
    "title": "OpenApi",
    "version": Any<String>,
  },
  "openapi": "3.0.1",
  "paths": Object {
    "/{hello}/advanced": Object {
      "post": Object {
        "parameters": Array [
          Object {
            "in": "path",
            "name": "hello",
            "required": true,
            "schema": Object {
              "type": "string",
            },
          },
        ],
        "requestBody": Object {
          "content": Object {
            "application/json": Object {
              "schema": Object {
                "$ref": "#/components/schemas/AdvancedModel",
              },
            },
          },
          "required": true,
        },
        "responses": Object {
          "200": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "$ref": "#/components/schemas/ResponseModel",
                },
              },
            },
            "description": "200 response",
            "headers": Object {
              "Access-Control-Allow-Credentials": Object {
                "schema": Object {
                  "type": "string",
                },
              },
              "Access-Control-Allow-Origin": Object {
                "schema": Object {
                  "type": "string",
                },
              },
              "Content-Type": Object {
                "schema": Object {
                  "type": "string",
                },
              },
            },
          },
        },
        "security": Array [
          Object {
            "authorizer": Array [],
          },
        ],
      },
    },
    "/{hello}/basic": Object {
      "post": Object {
        "parameters": Array [
          Object {
            "in": "path",
            "name": "hello",
            "required": true,
            "schema": Object {
              "type": "string",
            },
          },
        ],
        "requestBody": Object {
          "content": Object {
            "application/json": Object {
              "schema": Object {
                "$ref": "#/components/schemas/BasicModel",
              },
            },
          },
          "required": true,
        },
        "responses": Object {
          "200": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "$ref": "#/components/schemas/ResponseModel",
                },
              },
            },
            "description": "200 response",
            "headers": Object {
              "Access-Control-Allow-Credentials": Object {
                "schema": Object {
                  "type": "string",
                },
              },
              "Access-Control-Allow-Origin": Object {
                "schema": Object {
                  "type": "string",
                },
              },
              "Content-Type": Object {
                "schema": Object {
                  "type": "string",
                },
              },
            },
          },
        },
        "security": Array [
          Object {
            "authorizer": Array [],
          },
        ],
      },
    },
  },
}
`;

exports[`Snapshot 1`] = `
Object {
  "Outputs": Object {
    "OpenApiEndpointAD04A46A": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "OpenApi667EDE0B",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "OpenApiDeploymentStageprod3BBCA9DE",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "AssetParameters4229e3ca613f457841c56f01ac9c10a33a209ecd00ed87c06aa628b1d1063b76ArtifactHashF467ECE7": Object {
      "Description": "Artifact hash for asset \\"4229e3ca613f457841c56f01ac9c10a33a209ecd00ed87c06aa628b1d1063b76\\"",
      "Type": "String",
    },
    "AssetParameters4229e3ca613f457841c56f01ac9c10a33a209ecd00ed87c06aa628b1d1063b76S3Bucket38E835CA": Object {
      "Description": "S3 bucket for asset \\"4229e3ca613f457841c56f01ac9c10a33a209ecd00ed87c06aa628b1d1063b76\\"",
      "Type": "String",
    },
    "AssetParameters4229e3ca613f457841c56f01ac9c10a33a209ecd00ed87c06aa628b1d1063b76S3VersionKey2EDD6DB1": Object {
      "Description": "S3 key for asset version \\"4229e3ca613f457841c56f01ac9c10a33a209ecd00ed87c06aa628b1d1063b76\\"",
      "Type": "String",
    },
    "AssetParameters85328f19b68253a2c5a78f85945770d92a8e35594f71eb10e9b33acafa868726ArtifactHash9888BC96": Object {
      "Description": "Artifact hash for asset \\"85328f19b68253a2c5a78f85945770d92a8e35594f71eb10e9b33acafa868726\\"",
      "Type": "String",
    },
    "AssetParameters85328f19b68253a2c5a78f85945770d92a8e35594f71eb10e9b33acafa868726S3Bucket18A54A74": Object {
      "Description": "S3 bucket for asset \\"85328f19b68253a2c5a78f85945770d92a8e35594f71eb10e9b33acafa868726\\"",
      "Type": "String",
    },
    "AssetParameters85328f19b68253a2c5a78f85945770d92a8e35594f71eb10e9b33acafa868726S3VersionKeyFEE45B1A": Object {
      "Description": "S3 key for asset version \\"85328f19b68253a2c5a78f85945770d92a8e35594f71eb10e9b33acafa868726\\"",
      "Type": "String",
    },
    "AssetParametersed882f74d49503232e12bf9cada04c32c5c07a9222b6d2b06c5f23a481e84ab6ArtifactHash70C86D71": Object {
      "Description": "Artifact hash for asset \\"ed882f74d49503232e12bf9cada04c32c5c07a9222b6d2b06c5f23a481e84ab6\\"",
      "Type": "String",
    },
    "AssetParametersed882f74d49503232e12bf9cada04c32c5c07a9222b6d2b06c5f23a481e84ab6S3Bucket65B490D4": Object {
      "Description": "S3 bucket for asset \\"ed882f74d49503232e12bf9cada04c32c5c07a9222b6d2b06c5f23a481e84ab6\\"",
      "Type": "String",
    },
    "AssetParametersed882f74d49503232e12bf9cada04c32c5c07a9222b6d2b06c5f23a481e84ab6S3VersionKeyBA2A8C50": Object {
      "Description": "S3 key for asset version \\"ed882f74d49503232e12bf9cada04c32c5c07a9222b6d2b06c5f23a481e84ab6\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "OpenApi667EDE0B": Object {
      "Properties": Object {
        "Name": "OpenApi",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "OpenApiAccount550EBAC4": Object {
      "DependsOn": Array [
        "OpenApi667EDE0B",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "OpenApiCloudWatchRole86E92353",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "OpenApiAdvancedModelF9EAD729": Object {
      "Properties": Object {
        "ContentType": "application/json",
        "Name": "AdvancedModel",
        "RestApiId": Object {
          "Ref": "OpenApi667EDE0B",
        },
        "Schema": Object {
          "$schema": "http://json-schema.org/draft-07/schema#",
          "additionalProperties": false,
          "properties": Object {
            "basic": Object {
              "$ref": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "https://apigateway.amazonaws.com/restapis/",
                    Object {
                      "Ref": "OpenApi667EDE0B",
                    },
                    "/models/BasicModel",
                  ],
                ],
              },
            },
            "greeting": Object {
              "type": "string",
            },
            "postfix": Object {
              "type": "string",
            },
          },
          "required": Array [
            "greeting",
            "basic",
          ],
          "title": "AdvancedModel",
          "type": "object",
        },
      },
      "Type": "AWS::ApiGateway::Model",
    },
    "OpenApiBasicModel9C5F67F6": Object {
      "Properties": Object {
        "ContentType": "application/json",
        "Name": "BasicModel",
        "RestApiId": Object {
          "Ref": "OpenApi667EDE0B",
        },
        "Schema": Object {
          "$schema": "http://json-schema.org/draft-07/schema#",
          "additionalProperties": false,
          "properties": Object {
            "someNumber": Object {
              "type": "number",
            },
            "someString": Object {
              "type": "string",
            },
          },
          "required": Array [
            "someString",
            "someNumber",
          ],
          "title": "BasicModel",
          "type": "object",
        },
      },
      "Type": "AWS::ApiGateway::Model",
    },
    "OpenApiCloudWatchRole86E92353": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "OpenApiDeployment46B898978a783031191b66b80db5aaef44577382": Object {
      "DependsOn": Array [
        "OpenApiAdvancedModelF9EAD729",
        "OpenApiBasicModel9C5F67F6",
        "OpenApihelloadvancedPOSTAD3289AF",
        "OpenApihelloadvanced433A9690",
        "OpenApihellobasicPOSTF675B7C6",
        "OpenApihellobasic064FEE77",
        "OpenApihello8A8BE927",
        "OpenApiparamBodyValidator2BE5DCEC",
        "OpenApiResponseModel05515321",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "OpenApi667EDE0B",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "OpenApiDeploymentStageprod3BBCA9DE": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "OpenApiDeployment46B898978a783031191b66b80db5aaef44577382",
        },
        "RestApiId": Object {
          "Ref": "OpenApi667EDE0B",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "OpenApiResponseModel05515321": Object {
      "Properties": Object {
        "ContentType": "application/json",
        "Name": "ResponseModel",
        "RestApiId": Object {
          "Ref": "OpenApi667EDE0B",
        },
        "Schema": Object {
          "$schema": "http://json-schema.org/draft-07/schema#",
          "additionalProperties": false,
          "properties": Object {
            "message": Object {
              "type": "string",
            },
          },
          "required": Array [
            "message",
          ],
          "title": "ResponseModel",
          "type": "object",
        },
      },
      "Type": "AWS::ApiGateway::Model",
    },
    "OpenApihello8A8BE927": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "OpenApi667EDE0B",
            "RootResourceId",
          ],
        },
        "PathPart": "{hello}",
        "RestApiId": Object {
          "Ref": "OpenApi667EDE0B",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "OpenApihelloadvanced433A9690": Object {
      "Properties": Object {
        "ParentId": Object {
          "Ref": "OpenApihello8A8BE927",
        },
        "PathPart": "advanced",
        "RestApiId": Object {
          "Ref": "OpenApi667EDE0B",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "OpenApihelloadvancedPOSTAD3289AF": Object {
      "Properties": Object {
        "AuthorizationType": "CUSTOM",
        "AuthorizerId": Object {
          "Ref": "blogAuthorizerF1D3AEF3",
        },
        "HttpMethod": "POST",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "advancedLambdaFunction225720E0",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "MethodResponses": Array [
          Object {
            "ResponseModels": Object {
              "application/json": Object {
                "Ref": "OpenApiResponseModel05515321",
              },
            },
            "ResponseParameters": Object {
              "method.response.header.Access-Control-Allow-Credentials": true,
              "method.response.header.Access-Control-Allow-Origin": true,
              "method.response.header.Content-Type": true,
            },
            "StatusCode": "200",
          },
        ],
        "RequestModels": Object {
          "application/json": Object {
            "Ref": "OpenApiAdvancedModelF9EAD729",
          },
        },
        "RequestParameters": Object {
          "method.request.path.hello": true,
        },
        "RequestValidatorId": Object {
          "Ref": "OpenApiparamBodyValidator2BE5DCEC",
        },
        "ResourceId": Object {
          "Ref": "OpenApihelloadvanced433A9690",
        },
        "RestApiId": Object {
          "Ref": "OpenApi667EDE0B",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "OpenApihelloadvancedPOSTApiPermissionTesttestOpenApi1D4E6227POSThelloadvancedD4D12389": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "advancedLambdaFunction225720E0",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "OpenApi667EDE0B",
              },
              "/test-invoke-stage/POST/*/advanced",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "OpenApihelloadvancedPOSTApiPermissiontestOpenApi1D4E6227POSThelloadvanced22497703": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "advancedLambdaFunction225720E0",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "OpenApi667EDE0B",
              },
              "/",
              Object {
                "Ref": "OpenApiDeploymentStageprod3BBCA9DE",
              },
              "/POST/*/advanced",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "OpenApihellobasic064FEE77": Object {
      "Properties": Object {
        "ParentId": Object {
          "Ref": "OpenApihello8A8BE927",
        },
        "PathPart": "basic",
        "RestApiId": Object {
          "Ref": "OpenApi667EDE0B",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "OpenApihellobasicPOSTApiPermissionTesttestOpenApi1D4E6227POSThellobasicC19B13AB": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "basicLambdaFunctionF69553F9",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "OpenApi667EDE0B",
              },
              "/test-invoke-stage/POST/*/basic",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "OpenApihellobasicPOSTApiPermissiontestOpenApi1D4E6227POSThellobasic38FB83EC": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "basicLambdaFunctionF69553F9",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "OpenApi667EDE0B",
              },
              "/",
              Object {
                "Ref": "OpenApiDeploymentStageprod3BBCA9DE",
              },
              "/POST/*/basic",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "OpenApihellobasicPOSTF675B7C6": Object {
      "Properties": Object {
        "AuthorizationType": "CUSTOM",
        "AuthorizerId": Object {
          "Ref": "blogAuthorizerF1D3AEF3",
        },
        "HttpMethod": "POST",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "basicLambdaFunctionF69553F9",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "MethodResponses": Array [
          Object {
            "ResponseModels": Object {
              "application/json": Object {
                "Ref": "OpenApiResponseModel05515321",
              },
            },
            "ResponseParameters": Object {
              "method.response.header.Access-Control-Allow-Credentials": true,
              "method.response.header.Access-Control-Allow-Origin": true,
              "method.response.header.Content-Type": true,
            },
            "StatusCode": "200",
          },
        ],
        "RequestModels": Object {
          "application/json": Object {
            "Ref": "OpenApiBasicModel9C5F67F6",
          },
        },
        "RequestParameters": Object {
          "method.request.path.hello": true,
        },
        "RequestValidatorId": Object {
          "Ref": "OpenApiparamBodyValidator2BE5DCEC",
        },
        "ResourceId": Object {
          "Ref": "OpenApihellobasic064FEE77",
        },
        "RestApiId": Object {
          "Ref": "OpenApi667EDE0B",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "OpenApiparamBodyValidator2BE5DCEC": Object {
      "Properties": Object {
        "RestApiId": Object {
          "Ref": "OpenApi667EDE0B",
        },
        "ValidateRequestBody": true,
        "ValidateRequestParameters": true,
      },
      "Type": "AWS::ApiGateway::RequestValidator",
    },
    "advancedLambdaFunction225720E0": Object {
      "DependsOn": Array [
        "advancedLambdaFunctionServiceRole94547389",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersed882f74d49503232e12bf9cada04c32c5c07a9222b6d2b06c5f23a481e84ab6S3Bucket65B490D4",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersed882f74d49503232e12bf9cada04c32c5c07a9222b6d2b06c5f23a481e84ab6S3VersionKeyBA2A8C50",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersed882f74d49503232e12bf9cada04c32c5c07a9222b6d2b06c5f23a481e84ab6S3VersionKeyBA2A8C50",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "advancedLambdaFunctionServiceRole94547389",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "advancedLambdaFunctionServiceRole94547389": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "authorizerLambdaFunction52301750": Object {
      "DependsOn": Array [
        "authorizerLambdaFunctionServiceRole2AEEDAD3",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters85328f19b68253a2c5a78f85945770d92a8e35594f71eb10e9b33acafa868726S3Bucket18A54A74",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters85328f19b68253a2c5a78f85945770d92a8e35594f71eb10e9b33acafa868726S3VersionKeyFEE45B1A",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters85328f19b68253a2c5a78f85945770d92a8e35594f71eb10e9b33acafa868726S3VersionKeyFEE45B1A",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
          },
        },
        "FunctionName": "blogAuthorizer",
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "authorizerLambdaFunctionServiceRole2AEEDAD3",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "authorizerLambdaFunctionServiceRole2AEEDAD3": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "authorizerLambdaFunctiontestblogAuthorizerD3C51414Permissions73ABAB10": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "authorizerLambdaFunction52301750",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "OpenApi667EDE0B",
              },
              "/authorizers/",
              Object {
                "Ref": "blogAuthorizerF1D3AEF3",
              },
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "basicLambdaFunctionF69553F9": Object {
      "DependsOn": Array [
        "basicLambdaFunctionServiceRole3D0D875F",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters4229e3ca613f457841c56f01ac9c10a33a209ecd00ed87c06aa628b1d1063b76S3Bucket38E835CA",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters4229e3ca613f457841c56f01ac9c10a33a209ecd00ed87c06aa628b1d1063b76S3VersionKey2EDD6DB1",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters4229e3ca613f457841c56f01ac9c10a33a209ecd00ed87c06aa628b1d1063b76S3VersionKey2EDD6DB1",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "basicLambdaFunctionServiceRole3D0D875F",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "basicLambdaFunctionServiceRole3D0D875F": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "blogAuthorizerF1D3AEF3": Object {
      "Properties": Object {
        "AuthorizerUri": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":apigateway:",
              Object {
                "Ref": "AWS::Region",
              },
              ":lambda:path/2015-03-31/functions/",
              Object {
                "Fn::GetAtt": Array [
                  "authorizerLambdaFunction52301750",
                  "Arn",
                ],
              },
              "/invocations",
            ],
          ],
        },
        "IdentitySource": "method.request.header.Authorization",
        "Name": "testblogAuthorizerD3C51414",
        "RestApiId": Object {
          "Ref": "OpenApi667EDE0B",
        },
        "Type": "TOKEN",
      },
      "Type": "AWS::ApiGateway::Authorizer",
    },
  },
}
`;
